src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -999,23 +1011,34 @@ private bool IsAmbiguousDynamicParamsArgument<TMethodOrPropertySymbol>(ArrayBuil
                 return null;
             }
 
-            var resultWithSingleCandidate = OverloadResolutionResult<MethodSymbol>.GetInstance();
-            resultWithSingleCandidate.ResultsBuilder.Add(methodResolutionResult);
+            // For C# 12 and earlier statically bind invocations in presence of dynamic arguments only for local functions, extension methods or expanded non-array params cases.
+            if (Compilation.LanguageVersion > LanguageVersion.CSharp12 ||
+                singleCandidate.MethodKind == MethodKind.LocalFunction ||
+                resolution.IsExtensionMethodGroup ||
```

Comment:
Had a brief chat with Mads. He said LDM cares about the QB fix to the extent that it affects C# 12 and older behavior. 
Aside from the exception of local functions, he thought we should actually dynamify more. So it's not clear that just statically binding extension methods is the desired behavior for C# 12 (or 13).

Reply:
I wasn't sure what was meant by dynamify in this context.

Reply:
Yes, sorry for lack of clarity. With the known exception of local functions, he thought we should generally get a `dynamic` result back.

Reply:
I am not sure what, if anything, is suggested here. Extension methods cannot be invoked dynamically. See ```if (resolution.IsExtensionMethodGroup)``` on line 833.

Reply:
From offline discussion, making extension methods error is safer behavior for C# 12.